[docs] configuration.md - grammar & word-smithing (#9938)
expo/expo: docs/pages/workflow/configuration.md

### Configuration Resolution Rules

There are two different types of configs: static (`app.config.json`, `app.json`), and dynamic (`app.config.js`, `app.config.ts`). Static configs can be automatically updated with CLI tools, whereas dynamic configs must be manually updated by the developer.

> 1. The static config are read if `app.config.json` exists (falls back to `app.json`). If no static config exists, then default values are inferred from the `package.json` and your dependencies.

Reviewer: ```suggestion
1. The static config is read if `app.config.json` exists (falls back to `app.json`). If no static config exists, then default values are inferred from the `package.json` and your dependencies.
```

> 2. The dynamic config are read if either `app.config.ts` or `app.config.js` exist. If both exist, then the TypeScript config is used.

Reviewer: ```suggestion
2. The dynamic config is read if either `app.config.ts` or `app.config.js` exist. If both exist, then the TypeScript config is used.
```

> There are two different types of configs static (`app.config.json`, `app.json`) and dynamic (`app.config.js`, `app.config.ts`). Static configs can be automatically updated with CLI tools, whereas dynamic configs must be manually updated by the developer.

Reviewer: ```suggestion
There are two different types of configs: static (`app.config.json`, `app.json`), and dynamic (`app.config.js`, `app.config.ts`). Static configs can be automatically updated with CLI tools, whereas dynamic configs must be manually updated by the developer.
```

> 1. First the static config will be read if `app.config.json` exists, then falling back to `app.json`. If no static config exists then default values are inferred from the `package.json` and installed node modules.
> 2. The dynamic config will read if `app.config.ts` or `app.config.js` exist. If both exist then the TypeScript config is used.
> 3. If the dynamic config returns a function, then the static config is passed to the function with `({ config }) => ({})`. This function can mutate the static config values.
> 4. Whatever is returned from the dynamic config is used as the final config. It cannot have any promises.
> 5. All functions in the config are evaluated and serialized before any tool in the Expo ecosystem uses it. The config must be a JSON manifest when it is hosted.

Reviewer: ```suggestion
1. The static config will be read if `app.config.json` exists (falls back to `app.json`). If no static config exists, then default values are inferred from the `package.json` and your dependencies.
2. The dynamic config will be read if either `app.config.ts` or `app.config.js` exist. If both exist, then the TypeScript config is used.
3. If the dynamic config returns a function, then the static config is passed to the function with `({ config }) => ({})`. This function can then mutate the static config values.
4. Whatever is returned from the dynamic config is used as the final config. It cannot have any promises.
5. All functions in the config are evaluated and serialized before any tool in the Expo ecosystem uses it. The config must be a JSON manifest when it is hosted.
```
